### PR TITLE
Fix cd by adding to $PATH

### PR DESCRIPTION
**Pull request by Zha0q1**
https://github.com/apache/incubator-mxnet/issues/19917
This pr tries to fix ^ this issue. Joe helped update g4 instance to use the new nvidia-460 driver, which should be higher than the minimum requirement of the cu112 docker

### CODE REVIEW

**Review by josephevans — APPROVED** [review]

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -1307,13 +1307,14 @@ cd_pypi_publish() {
...
     set -ex
-    pip3 install --user awscli
+    pip3 install --upgrade --user awscli
     filepath=$(readlink -f wheel_build/dist/*.whl)
     filename=$(basename $filepath)
     variant=$(echo $filename | cut -d'-' -f1 | cut -d'_' -f2 -s)
     if [ -z "${variant}" ]; then
         variant="cpu"
     fi
+    export PATH=/usr/local/bin:$PATH
```
Comment (leezu) [inline ci/docker/runtime_functions.sh:1317]: Do you mean that `/usr/local/bin` wasn't on the PATH before? Do you know why it worked at all before?

Comment (Zha0q1) [reply ci/docker/runtime_functions.sh:1317]: We recently changed the ami used by restricted-mxnetlinux-gpu to update the nvidia driver

Comment (Zha0q1) [reply ci/docker/runtime_functions.sh:1317]: Do you think if there is a better place to set PATH?

Comment (Zha0q1) [reply ci/docker/runtime_functions.sh:1317]: @leezu

Comment (leezu) [reply ci/docker/runtime_functions.sh:1317]: It's fine to add it here, I'm just trying to understand the reasoning. Can you explain how it's related to AMI? Wouldn't `cd_s3_publish` run in a docker container and be independent from the AMI?

**Pull request merged**